http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java

```diff
@@ -30,25 +30,42 @@
  */
 @SdkProtectedApi
 public final class SdkHttpConfigurationOption<T> extends AttributeMap.Key<T> {
-
     /**
      * Timeout for each read to the underlying socket.
      */
-    public static final SdkHttpConfigurationOption<Duration> SOCKET_TIMEOUT =
-            new SdkHttpConfigurationOption<>("SocketTimeout", Duration.class);
+    public static final SdkHttpConfigurationOption<Duration> SOCKET_READ_TIMEOUT =
+            new SdkHttpConfigurationOption<>("SocketReadTimeout", Duration.class);
+
+    /**
+     * Timeout for each write to the underlying socket.
+     */
+    public static final SdkHttpConfigurationOption<Duration> SOCKET_WRITE_TIMEOUT =
+            new SdkHttpConfigurationOption<>("SocketWriteTimeout", Duration.class);
 
     /**
      * Timeout for establishing a connection to a remote service.
      */
     public static final SdkHttpConfigurationOption<Duration> CONNECTION_TIMEOUT =
             new SdkHttpConfigurationOption<>("ConnectionTimeout", Duration.class);
 
+    /**
+     * Timeout for acquiring an already-established connection from a connection pool to a remote service.
+     */
+    public static final SdkHttpConfigurationOption<Duration> CONNECTION_ACQUIRE_TIMEOUT =
```

Comment:
Whats our stance for adding configuration options to be supplied by the runtime? I worry about coupling ourselves to tightly with the netty implementation by including too many options when the implementation itself can choose reasonable defaults. We would definitely want settings for those that may vary service to service like SO timeouts, max conns, etc.

Reply:
I agree. I think we should use it for:
1. Things we want services to be able to adjust
2. Things we want to provide sensible defaults for that could apply widely across plugin implementations
3. Things we would want all general-use HTTP plugins to try to support, because they're useful features to have

There are other advantages to including more here that I don't find quite as compelling, like centralizing common defaults so that we can prevent divergence between the clients that we own, giving customers one place to look to find our defaults, and creating a common verbiage to refer to the same features.